[2024/08/19] Feat/camp-test 캠프 테스트 코드 추가

gminnimk opened the pull request:
## 📌 What is this PR ?

캠프 테스트 코드 추가

<br/>

## 🔎 Additions / Changes

Additions

- createCamp,

- updateCampRankingsBasedOnLikes  

- 테스트 코드 추가 

Changes

- jwt 하드 코딩 개선 

- 캠프 서비스 예외 하드 코딩 메시지 개선

<br/>

## 📷 Screenshot

| 테스트 코드 | 스크린샷 |
| --- | --- |
| createCamp | ![image](https://github.com/user-attachments/assets/902db611-16cc-43ad-a352-fd3124a61dfe) |
| updateCampRankingsBasedOnLikes | ![image](https://github.com/user-attachments/assets/f213efa7-a96d-4ebc-9f60-3452e84e1287) |

<br/>

gminnimk committed:
🩹 fix : #99 수정 가능한 리스트로 수정

gminnimk committed:
🍻 test : #99 creatCamp, updateCampRankingsBasedOnLikes 테스트 코드 작성

gminnimk committed:
♻️ refactor : #99 예외 메시지 하드 코딩 개선

gminnimk committed:
♻️ refactor : #99 jwt 하드 코딩 개선

thesun4sky commented on src/test/java/com/sparta/studytrek/domain/camp/service/CampServiceTest.java:
```diff
@@ -0,0 +1,161 @@
+        assertEquals(description, response.description());
+        assertEquals(imageUrl, response.imageUrl());
+
+        verify(campRepository, times(1)).findByName(campName);
```
verify(campRepository, times(1)).findByName(eq(campName));

이런식으로 eq() 메서드를 사용해보세요

gminnimk replied:
감사드립니다! 🙇🏼‍♂️

thesun4sky commented on src/test/java/com/sparta/studytrek/domain/camp/service/CampServiceTest.java:
```diff
@@ -0,0 +1,161 @@
+        });
+
+        // Then
+        assertEquals("이미지 파일이 필요합니다.", exception.getMessage());
```
"이미지 파일이 필요합니다." 값은 상수값으로 추출해서 해당 변수를 참조하도록 수정해주세요.

ex.
```
IMAGE_NULL_EXCEPTION_MESSAGE = "이미지 파일이 필요합니다."
```

gminnimk replied:
네! 확인했습니다!

thesun4sky commented on src/test/java/com/sparta/studytrek/domain/camp/service/CampServiceTest.java:
```diff
@@ -0,0 +1,161 @@
+        // When
+        campService.updateCampRankingsBasedOnLikes();
+
+        // Then
```
then 체크항목에서 아래 코드에서 수행된 campRank 값이 수정된걸 조회해서 확인해주어야 합니다.

https://github.com/StUdY-TrEk/Boot-Camping/blob/37ac2cc83f34259048d19faaa13657eac928f49b/src/main/java/com/sparta/studytrek/domain/camp/service/CampService.java#L128C13-L128C21

gminnimk replied:
네! 확인했습니다! 감사드립니다!!

thesun4sky reviewed (commented):
리뷰 확인 부탁드립니다.

gminnimk committed:
🍻 test : #99 기존 테스트 코드 수정

- eq 적용
- 하드 코딩 개선
- testUpdateCampRankingsBasedOnLikes 메서드 then 아래 campRank 값이 수정된걸 조회
